pypdf/filters.py

```diff
@@ -340,7 +341,8 @@ def decode(
         index = 0
         while True:
             if index >= len(data):
-                raise PdfStreamError("Unexpected EOD in RunLengthDecode")
+                break  # reach End Of String even if no EOD
```

Comment:
Same question here: Does this have any side-effect? Should we add a warning / a debug log message?

Reply:
same as above